frontend/front/src/pages/Public/Pages/SurveyPage.js

```diff
@@ -76,14 +79,33 @@ export const SurveyPage = () => {
     [addSurveyVisit, getReCaptchaToken]
   );
 
+  const isCanonicalized = validationStatus === "canonicalized";
+
   const step = useMemo(() => {
-    if (!createHomeData && !isSurveyVisitSuccess) {
+    if (!isCanonicalized && !isSurveyVisitSuccess) {
       return STEP_ADDRESS;
     } else if (isSurveyVisitSuccess) {
       return STEP_THANKS;
+    } else if (!!createHomeData && isCanonicalized) {
+      return STEP_SURVEY;
```

Comment:
Removed `!createHomeData` because we want to keep the AddressValidator component visible while canonicalization takes place, which keeps the form filled in. Once canonicalized, `step` can move on to `STEP_SURVEY`.